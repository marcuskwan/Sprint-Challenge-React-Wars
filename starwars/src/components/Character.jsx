import React from "react";
import { Card, Segment, Dimmer, Loader, Image } from "semantic-ui-react";
import PropTypes from "prop-types";

function Character({ characterData }) {
  return (
    <div className="character">
      {!characterData ? (
        <Segment>
          <Dimmer active inverted>
            <Loader inverted>Loading</Loader>
          </Dimmer>

          <Image src="/images/wireframe/short-paragraph.png" />
        </Segment>
      ) : (
        <Card centered>
          <Card.Content>
            <Card.Header>{characterData.name}</Card.Header>
          </Card.Content>
          <Card.Content>
            <Card.Meta>{characterData.birth_year}</Card.Meta>
            <Card.Meta>{characterData.eye_color}</Card.Meta>
            <Card.Meta>{characterData.birth_year}</Card.Meta>
            <Card.Meta>{characterData.gender}</Card.Meta>
            <Card.Meta>{characterData.hair_color}</Card.Meta>
            <Card.Meta>{characterData.height}</Card.Meta>
            <Card.Meta>{characterData.mass}</Card.Meta>
            <Card.Meta>{characterData.skin_color}</Card.Meta>
          </Card.Content>
        </Card>
      )}
    </div>
  );
}

Character.propTypes = {
  characterData: PropTypes.shape({
    birth_year: PropTypes.string,
    created: PropTypes.string,
    edited: PropTypes.string,
    eye_color: PropTypes.string,
    films: PropTypes.arrayOf(PropTypes.string),
    gender: PropTypes.string,
    hair_color: PropTypes.string,
    height: PropTypes.string,
    homeworld: PropTypes.string,
    mass: PropTypes.string,
    name: PropTypes.string,
    skin_color: PropTypes.string,
    species: PropTypes.arrayOf(PropTypes.string),
    starships: PropTypes.arrayOf(PropTypes.string),
    url: PropTypes.string,
    vehicles: PropTypes.arrayOf(PropTypes.string),
  }),
};

export default Character;
