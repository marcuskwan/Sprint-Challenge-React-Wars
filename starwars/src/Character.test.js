import React from "react";

import renderer from "react-test-renderer";
import Character from "./components/Character";

describe("<Character />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Character />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
