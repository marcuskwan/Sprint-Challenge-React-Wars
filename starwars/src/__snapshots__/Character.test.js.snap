// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> should match snapshot 1`] = `
<div
  className="character"
>
  <div
    className="ui segment"
  >
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted text loader"
        >
          Loading
        </div>
      </div>
    </div>
    <img
      className="ui image"
      src="/images/wireframe/short-paragraph.png"
    />
  </div>
</div>
`;
